amundsen_application/base/base_mail_client.py

```diff
@@ -11,19 +11,19 @@ def __init__(self, recipients: List[str]) -> None:
 
     @abc.abstractmethod
     def send_email(self,
-                   sender: str,
-                   recipients: List[str],
                    subject: str,
                    text: str,
                    html: str,
+                   sender: str,
```

Comment:
nit: It looks like you had to change this because you changed the order of how things were called in `notification_utils.py`. As a cleanup, if we switch using keyword args we won't have to worry about parameter order. Might as well have them alphabetical as well.
```
def send_email(self, *, subject: str, ....)
```